home/import_helpers.py

```diff
@@ -165,21 +165,27 @@ def extract_errors(data: dict[str | int, Any] | list[str]) -> dict[str, str]:
     return error_message
 
 
-def fix_rows(
-    rows: Generator[dict[str, str], None, None] | Iterator[dict[str | Any, Any]]
-) -> Iterator[dict[str, str | None]]:
+def fix_rows(rows: Iterator[dict[str | Any, Any]]) -> Iterator[dict[str, str | None]]:
     """
     Fix keys for all rows by lowercasing keys and removing whitespace from keys and values
     """
-    for row in rows:
-        yield fix_row(row)
+    for index, row in enumerate(rows):
+        yield fix_row(row, index)
 
 
-def fix_row(row: dict[str, str]) -> dict[str, str | None]:
+def fix_row(row: dict[str, str | None], index: int) -> dict[str, str | None]:
     """
     Fix a single row by lowercasing the key and removing whitespace from the key and value
     """
     try:
+        if index == 0:
+            keys = [_normalise_key(k) for k, v in row.items()]
+            if len(keys) != len(set(keys)):
+                raise ImportException(
+                    "Invalid format. Please check that there are no duplicate headers.",
+                    row_num=index,
+                )
+
```

Comment:
Could we do this check up in `fix_rows()` with `if len(row) != len(fix_row(row))` on the first iteration? Perhaps with `first_row = next(rows)` before the loop?

Reply:
Done